subprojects/core/src/main/java/org/gradle/cache/internal/GradleUserHomeCacheCleanupEnablement.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cache.internal;
+
+import org.gradle.cache.CleanupAction;
+import org.gradle.initialization.GradleUserHomeDirProvider;
+import org.gradle.util.internal.GUtil;
+
+import java.io.File;
+import java.util.Properties;
+
+public class GradleUserHomeCacheCleanupEnablement implements CacheCleanupEnablement {
+    public static final String CACHE_CLEANUP_PROPERTY = "org.gradle.cache.cleanup";
+
+    private final GradleUserHomeDirProvider userHomeDirProvider;
+
+    public GradleUserHomeCacheCleanupEnablement(GradleUserHomeDirProvider userHomeDirProvider) {
+        this.userHomeDirProvider = userHomeDirProvider;
+    }
+
+    private boolean isDisabled() {
```

Comment:
I thought it was easier to reason about this way, since you disable the capability instead of enable it, but I'm not sure that's a compelling reason for it to exist.  I've changed it.

Reply:
> since you disable the capability instead of enable it,

Ah, ok.  I didn't pick up on that.  I still think it's better without it.